genomicsdb/scripts/genomicsdb_query.py

```diff
@@ -618,52 +625,85 @@ def process(config):
         logging.error(msg + f" not imported into workspace({export_config.workspace})")
         return -1
     global gdb
-    try:
-        if gdb:
-            logging.info("Found gdb to process " + msg)
-        else:
-            logging.error("Something wrong, gdb seems to be None")
+    # Allow one retry to account for expired access tokens for azure URLs
+    if export_config.workspace.startswith("az://"):
+        allow_retry = True
+    else:
+        allow_retry = False
+    while True:
+        try:
+            if gdb:
+                logging.info("Found gdb to process " + msg)
+            else:
+                logging.info("Starting new gdb to process " + msg)
+                gdb = instantiate_genomicsdb(configure_export(export_config), msg)
+        except NameError:
+            gdb = instantiate_genomicsdb(configure_export(export_config), msg)
+
+        query_protobuf = configure_query(query_config)
+
+        try:
+            if output_config.type == "csv":
+                df = gdb.query_variant_calls(query_protobuf=query_protobuf, flatten_intervals=True)
+                df.to_csv(output_config.filename, index=False)
+            elif output_config.type == "json":
+                json_output = gdb.query_variant_calls(
+                    query_protobuf=query_protobuf, json_output=output_config.json_type
+                )
+                with open(output_config.filename, "wb") as f:
+                    f.write(json_output)
+            elif output_config.type == "arrow":
+                nbytes = 0
+                writer = None
+                i = 0
+                for out in gdb.query_variant_calls(query_protobuf=query_protobuf, arrow_output=True, batching=True):
+                    reader = pa.ipc.open_stream(out)
+                    for batch in reader:
+                        if nbytes > output_config.max_arrow_bytes:
+                            i += 1
+                            nbytes = 0
+                            if writer:
+                                writer.close()
+                                writer = None
+                        if not writer:
+                            writer = pq.ParquetWriter(f"{output_config.filename}__{i}.parquet", batch.schema)
+                        nbytes += batch.nbytes
+                        writer.write_batch(batch)
+                    if writer:
+                        writer.close()
+                        writer = None
+            logging.info(f"Processed {msg}")
+            return 0
+        except Exception as e:
+            if allow_retry:
+                # Check for the possibility of an expired access token
+                allow_retry = False
+                try:
+                    if not gdb.workspace_exists(export_config.workspace):
+                        logging.info(f"Retrying query with a new instance of gdb for {msg}...")
```

Comment:
Could you explain what we're trying to do with this check?

Reply:
I m trying to check if the gdb is functional. If it is, it is probably not an expired token. But, I am still allowing  a retry for now. I will document that better in the code. And maybe improve the logging message. Comments?